subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/CaptureStateBeforeExecutionStepTest.groovy

```diff
@@ -34,29 +32,28 @@ import org.gradle.internal.snapshot.FileSystemSnapshot
 import org.gradle.internal.snapshot.ValueSnapshot
 import org.gradle.internal.snapshot.ValueSnapshotter
 import org.gradle.internal.snapshot.impl.ImplementationSnapshot
-import spock.lang.Specification
 
 import static org.gradle.internal.execution.UnitOfWork.OverlappingOutputHandling.DETECT_OVERLAPS
 import static org.gradle.internal.execution.UnitOfWork.OverlappingOutputHandling.IGNORE_OVERLAPS
+import static org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.Operation.Result
 
-class CaptureStateBeforeExecutionStepTest extends Specification {
+class CaptureStateBeforeExecutionStepTest extends StepSpec {
 
     def classloaderHierarchyHasher = Mock(ClassLoaderHierarchyHasher)
     def valueSnapshotter = Mock(ValueSnapshotter)
-    def work = Mock(UnitOfWork)
-    def afterPreviousExecutionContext = Stub(AfterPreviousExecutionContext) {
-        getWork() >> this.work
-    }
-    Step<BeforeExecutionContext, CachingResult> delegate = Mock()
+    // Spock 1.3 fails when using the inherited Stub here
```

Comment:
OK, so it seems that when you configure a `Stub` using:

```groovy
        work.visitInputProperties(_ as UnitOfWork.InputPropertyVisitor)
```

...in _some_ cases it can throw an exception complaining:

```text
No signature of method: org.spockframework.lang.Wildcard.visitInputProperty() is applicable for argument types: (String, String) values: [inputString, myValue]
groovy.lang.MissingMethodException: No signature of method: org.spockframework.lang.Wildcard.visitInputProperty() is applicable for argument types: (String, String) values: [inputString, myValue]
```

This doesn't happen if you configure it by the wildcard cardinality:

```groovy
        _ * work.visitInputProperties(_ as UnitOfWork.InputPropertyVisitor)
```

I've now changed all stubs to be configured using the wildcard cardinality, and it fixed the problem.

Doing so makes the code more readable, too, as it is now easy to tell when a stub is being configured.